toolkit/tools/pkg/imagecustomizerlib/customizeboot.go

```diff
@@ -57,10 +60,179 @@ func handleKernelCommandLine(extraCommandLine string, imageChroot *safechroot.Ch
 	newGrub2ConfigFile := grub2ConfigFile[:insertIndex] + extraCommandLine + " " + grub2ConfigFile[insertIndex:]
 
 	// Update grub.cfg file.
-	err = os.WriteFile(grub2ConfigFilePath, []byte(newGrub2ConfigFile), 0)
+	err = writeGrub2ConfigFile(newGrub2ConfigFile, imageChroot)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Updates the kernel command-line args with the new SELinux mode.
+//
+// See, installutils.setGrubCfgSELinux()
+func updateSELinuxCommandLine(selinuxMode imagecustomizerapi.SELinux, imageChroot *safechroot.Chroot) error {
+	logger.Log.Infof("Updating SELinux kernel command-line args")
+
+	grub2Config, err := readGrub2ConfigFile(imageChroot)
+	if err != nil {
+		return err
+	}
+
+	// Try to find any existing SELinux args.
+	selinuxArgsMatch := linuxCommandLineSELinuxRegex.FindStringSubmatchIndex(grub2Config)
+
+	start := 0
+	end := 0
+
+	if selinuxArgsMatch == nil {
+		// SELinux args are not in the grub.cfg file.
+		if selinuxMode == imagecustomizerapi.SELinuxDisabled {
+			// Nothing to do.
+			return nil
+		}
+
+		insertLocationMatch := linuxCommandLineInsertRegex.FindStringIndex(grub2Config)
+		if insertLocationMatch == nil {
+			return fmt.Errorf("failed to find location to insert SELinux kernel command-line args")
+		}
+
+		// Use the end of the regex match as the point to do the insert at.
+		insertPoint := insertLocationMatch[1]
+		start = insertPoint
+
+		// -1 is a sneaky way of adding a whitespace character after the inserted SELinux args.
+		end = insertPoint - 1
+	} else {
+		// Get the range of the existing SELinux args.
+		// Note: regexp returns index pairs. So, [2] and [3] are the start and end indexes respectively of the 1st
+		// sub-match.
+		start = selinuxArgsMatch[2]
+		end = selinuxArgsMatch[3]
+	}
+
+	newSELinuxArgs := ""
+	switch selinuxMode {
+	case imagecustomizerapi.SELinuxForceEnforcing:
+		newSELinuxArgs = installutils.CmdlineSELinuxForceEnforcing
+
+	case imagecustomizerapi.SELinuxPermissive, imagecustomizerapi.SELinuxEnforcing:
+		newSELinuxArgs = installutils.CmdlineSELinuxSettings
+
+	case imagecustomizerapi.SELinuxDisabled:
+		newSELinuxArgs = ""
+
+	default:
+		return fmt.Errorf("unknown SELinux mode (%s)", selinuxMode)
+	}
+
+	// Insert/replace/delete SELinux command-line arguments.
+	newGrub2Config := grub2Config[:start] + newSELinuxArgs + grub2Config[end:]
+
+	// Update grub.cfg file.
+	err = writeGrub2ConfigFile(newGrub2Config, imageChroot)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getCurrentSELinuxMode(imageChroot *safechroot.Chroot) (imagecustomizerapi.SELinux, error) {
+	logger.Log.Debugf("Get existing SELinux mode")
+
+	grub2Config, err := readGrub2ConfigFile(imageChroot)
+	if err != nil {
+		return "", err
```

Comment:
Nit-pick: I'm aware it's just a string but since we already defined the `SELinux` type, could we use its named values instead of a raw empty string? For the sake of consistency and to be sure we always return an valid value (even if there's a good chance it won't be used).

Reply:
Hmm. `""` is a lot shorted than `SELinuxDefault`. This reduces the visual noise and IMHO makes it easier to read the error.

Reply:
Hmmm. In many places in the code we also follow the approach of named output, where you can then use a simple `return` without additional arguments. That might also help here to reduce the repetition completely.

Reply:
fixed